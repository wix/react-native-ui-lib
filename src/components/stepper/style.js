import {StyleSheet, PixelRatio} from 'react-native';
import {Constants} from '../../helpers';
import {Colors, Typography, ThemeManager} from '../../style';

export default function createStyles() {
  const separatorColor = Colors.dark70;
  const bottomTextMargin = 4; // use margins to center the text until we move to using assets
  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center'
      // paddingLeft: Constants.isIOS ? 16 : undefined,
      // paddingRight: Constants.isIOS ? 16 : undefined,
    },
    buttons: {
      flexDirection: 'row',
      borderRadius: Constants.isIOS ? 21 : 2,
      borderWidth: 1,
      borderColor: separatorColor,
      width: Constants.isIOS ? 100 : 98,
      height: Constants.isIOS ? 42 : 35
    },
    button: {
      flex: 0.5,
      alignItems: 'center',
      justifyContent: 'center'
    },
    title: {
      justifyContent: 'center',
      flex: 0.6,
      height: Constants.isIOS ? 70 : 68
    },
    titleText: {
      ...Typography.text70,
      color: Colors.dark10
    },
    separator: {
      marginTop: Constants.isIOS ? 4 : 2,
      height: Constants.isIOS ? 32 : 30,
      borderWidth: Constants.isIOS ? 1 / PixelRatio.get() : undefined,
      borderLeftWidth: Constants.isIOS ? undefined : 1,
      borderColor: separatorColor
    },
    buttonText: {
      ...Typography.text40,
      fontWeight: '200',
      color: ThemeManager.primaryColor,
      backgroundColor: 'transparent',
      marginBottom: bottomTextMargin
    },
    disableText: {
      color: Colors.dark70,
      marginBottom: bottomTextMargin
    }
  });
}
