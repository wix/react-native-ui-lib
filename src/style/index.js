module.exports = {
  get Typography() {
    return require('./typography').default;
  },
  get BorderRadiuses() {
    return require('./borderRadiuses').default;
  },
  get Colors() {
    return require('./colors').default;
  },
  get Shadows() {
    return require('./shadows').default;
  },
  get Spacings() {
    return require('./spacings').default;
  },
  get ComponentsColors() {
    return require('./componentsColors').default;
  },
  get Components() {
    return require('./components').default;
  },
  get ThemeManager() {
    return require('./themeManager').default;
  },
  get AnimatableManager() {
    return require('./animatableManager').default;
  }
};
