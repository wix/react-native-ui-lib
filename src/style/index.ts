export {default as Colors} from './colors';
export {default as DesignTokens} from './designTokens';
export {default as DesignTokensDM} from './designTokensDM';
export {default as Scheme, SchemeType, Schemes, SchemeChangeListener} from './scheme';
export {default as Typography} from './typography';
export {default as BorderRadiuses} from './borderRadiuses';
export {default as Shadows} from './shadows';
export {default as Spacings} from './spacings';
export {default as ComponentsColors} from './componentsColors';
export {default as Components} from './components';
export {default as ThemeManager} from './themeManager';
export {default as ColorName} from './colorName';
export {default as Dividers} from './dividers';
