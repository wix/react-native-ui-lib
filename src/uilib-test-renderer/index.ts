export * from '@testing-library/react-native';
import {findStyle} from './helper';
import {fireOnMomentumScrollEnd, fireOnScroll} from './scrollViewHelper';

export {
  findStyle,
  fireOnMomentumScrollEnd,
  fireOnScroll
};

/* Docs: https://callstack.github.io/react-native-testing-library/docs/api/  */
