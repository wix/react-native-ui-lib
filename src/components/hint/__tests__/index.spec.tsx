import React from 'react';
import {Hint} from 'react-native-ui-lib';
import {Colors} from '../../../style';
import {findStyle} from '../../../uilib-test-renderer';
import {HintDriver} from '../Hint.driver';

const HintTestComponent = ({
  showHint,
  color
}: {
  showHint: boolean;
  color?: string;
}) => {
  return (
    <Hint
      visible={showHint}
      message={'Hint message to hint things'}
      position={Hint.positions.TOP}
      useSideTip
      key={'1'}
      targetFrame={{x: 1, y: 1, height: 1, width: 1}}
      onBackgroundPress={() => {}}
      color={color}
      removePaddings
      enableShadow
      testID={'Hint'}
    />
  );
};

describe('Hint Screen component test', () => {
  afterEach(() => {
    HintDriver.clear();
  });

  it('Test Hint component background color', async () => {
    const expectedColor = Colors.$backgroundPrimaryHeavy;
    const component = <HintTestComponent showHint/>;

    const driver = new HintDriver({component, testID: 'Hint'});
    expect(await driver.getElement()).toBeTruthy();

    const hintContent = await driver.getHintContent();
    let color = findStyle('backgroundColor', hintContent);

    expect(color).toBe(expectedColor);
    expect(hintContent).toBeTruthy();

    const WhiteHint = <HintTestComponent showHint color={Colors.white}/>;
    const WhiteHintDriver = new HintDriver({component: WhiteHint, testID: 'Hint'});

    const WhiteHintContent = await WhiteHintDriver.getHintContent();
    color = findStyle('backgroundColor', WhiteHintContent);
    expect(color).toBe(Colors.white);
  });

  it('Test Hint modal is not visible when showHint is false', async () => {
    const component = <HintTestComponent showHint={false}/>;
    const driver = new HintDriver({component, testID: 'Hint'});
    expect(await driver.getHintModal()).toBeFalsy();
  });

  it('Test Hint modal is visible when showHint is true', async () => {
    const component = <HintTestComponent showHint/>;
    const driver = new HintDriver({component, testID: 'Hint'});
    expect(await driver.getElement()).toBeTruthy();
    expect(await driver.getHintModal()).toBeTruthy();
  });
});
