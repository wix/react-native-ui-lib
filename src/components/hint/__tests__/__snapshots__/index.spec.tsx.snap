// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint Screen component test Different positions and scenarios center position should position correctly for a target positioned on center - with sideTip 1`] = `
<View
  centerV={true}
  padding-s5={true}
  style={
    [
      undefined,
      undefined,
      undefined,
      {
        "flex": 1,
      },
      undefined,
      {
        "padding": 20,
      },
      {},
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "backgroundColor": undefined,
        "height": 1334,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
    testID="Hint.overlay"
  />
  <View
    collapsable={true}
    message="Hint message to hint things"
    modifiers={
      {
        "alignments": {},
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "color": undefined,
        "flexStyle": undefined,
        "margins": {},
        "paddings": {},
        "positionStyle": undefined,
        "typography": undefined,
      }
    }
    position="bottom"
    style={
      [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {},
        {},
        {},
        [
          {
            "position": "absolute",
          },
          undefined,
          {
            "left": 100,
            "top": 200,
          },
          {
            "elevation": 10,
            "zIndex": 10,
          },
        ],
      ]
    }
    targetFrame={
      {
        "height": 60,
        "width": 100,
        "x": 100,
        "y": 200,
      }
    }
    useModal={true}
    useSideTip={true}
    visible={true}
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        {
          "alignItems": "flex-start",
          "left": -100,
          "opacity": 1,
          "paddingVertical": 16,
          "position": "absolute",
          "top": 60,
          "transform": [
            {
              "translateY": 0,
            },
          ],
          "width": 750,
        }
      }
      testID="Hint"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          centerV={true}
          onLayout={[Function]}
          row={true}
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {},
              {},
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              [
                {
                  "backgroundColor": "#5A48F5",
                  "borderRadius": 20,
                  "maxWidth": 400,
                },
                {
                  "paddingBottom": 16,
                  "paddingHorizontal": 20,
                  "paddingTop": 12,
                },
                undefined,
                {
                  "backgroundColor": "#5A48F5",
                },
                false,
                {
                  "left": 0,
                },
              ],
            ]
          }
          testID="Hint.message"
        >
          <Text
            fsTagName="unmask"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "writingDirection": "ltr",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                [
                  {
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
            testID="Hint.message.text"
          >
            Hint message to hint things
          </Text>
        </View>
      </View>
      <Image
        accessibilityRole="image"
        accessible={false}
        assetGroup="icons"
        onError={[Function]}
        source={
          {
            "testUri": "../../../src/assets/internal/images/hintTipSide.png",
          }
        }
        style={
          [
            {
              "tintColor": "#5A48F5",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            false,
            [
              {
                "position": "absolute",
              },
              {
                "top": 9,
              },
              {
                "transform": [
                  {
                    "scaleY": 1,
                  },
                  {
                    "scaleX": 1,
                  },
                ],
              },
            ],
            false,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Hint Screen component test Different positions and scenarios center position should position correctly for a target positioned on center 1`] = `
<View
  centerV={true}
  padding-s5={true}
  style={
    [
      undefined,
      undefined,
      undefined,
      {
        "flex": 1,
      },
      undefined,
      {
        "padding": 20,
      },
      {},
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "backgroundColor": undefined,
        "height": 1334,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
    testID="Hint.overlay"
  />
  <View
    collapsable={true}
    message="Hint message to hint things"
    modifiers={
      {
        "alignments": {},
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "color": undefined,
        "flexStyle": undefined,
        "margins": {},
        "paddings": {},
        "positionStyle": undefined,
        "typography": undefined,
      }
    }
    position="bottom"
    style={
      [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {},
        {},
        {},
        [
          {
            "position": "absolute",
          },
          undefined,
          {
            "left": 100,
            "top": 200,
          },
          {
            "elevation": 10,
            "zIndex": 10,
          },
        ],
      ]
    }
    targetFrame={
      {
        "height": 60,
        "width": 100,
        "x": 100,
        "y": 200,
      }
    }
    useModal={true}
    visible={true}
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        {
          "alignItems": "flex-start",
          "left": -100,
          "opacity": 1,
          "paddingVertical": 16,
          "position": "absolute",
          "top": 60,
          "transform": [
            {
              "translateY": 0,
            },
          ],
          "width": 750,
        }
      }
      testID="Hint"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          centerV={true}
          onLayout={[Function]}
          row={true}
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {},
              {},
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              [
                {
                  "backgroundColor": "#5A48F5",
                  "borderRadius": 20,
                  "maxWidth": 400,
                },
                {
                  "paddingBottom": 16,
                  "paddingHorizontal": 20,
                  "paddingTop": 12,
                },
                undefined,
                {
                  "backgroundColor": "#5A48F5",
                },
                false,
                {
                  "left": 0,
                },
              ],
            ]
          }
          testID="Hint.message"
        >
          <Text
            fsTagName="unmask"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "writingDirection": "ltr",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                [
                  {
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
            testID="Hint.message.text"
          >
            Hint message to hint things
          </Text>
        </View>
      </View>
      <Image
        accessibilityRole="image"
        accessible={false}
        assetGroup="icons"
        onError={[Function]}
        source={
          {
            "testUri": "../../../src/assets/internal/images/hintTipMiddle.png",
          }
        }
        style={
          [
            {
              "tintColor": "#5A48F5",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            false,
            [
              {
                "position": "absolute",
              },
              {
                "top": 9,
              },
              {
                "transform": [
                  {
                    "scaleY": 1,
                  },
                  {
                    "scaleX": 1,
                  },
                ],
              },
            ],
            false,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Hint Screen component test Different positions and scenarios left position should position correctly for a target positioned on left - with sideTip 1`] = `
<View
  centerV={true}
  padding-s5={true}
  style={
    [
      undefined,
      undefined,
      undefined,
      {
        "flex": 1,
      },
      undefined,
      {
        "padding": 20,
      },
      {},
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "backgroundColor": undefined,
        "height": 1334,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
    testID="Hint.overlay"
  />
  <View
    collapsable={true}
    message="Hint message to hint things"
    modifiers={
      {
        "alignments": {},
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "color": undefined,
        "flexStyle": undefined,
        "margins": {},
        "paddings": {},
        "positionStyle": undefined,
        "typography": undefined,
      }
    }
    position="bottom"
    style={
      [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {},
        {},
        {},
        [
          {
            "position": "absolute",
          },
          undefined,
          {
            "left": 25,
            "top": 370,
          },
          {
            "elevation": 10,
            "zIndex": 10,
          },
        ],
      ]
    }
    targetFrame={
      {
        "height": 60,
        "width": 100,
        "x": 25,
        "y": 370,
      }
    }
    useModal={true}
    useSideTip={true}
    visible={true}
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        {
          "alignItems": "flex-start",
          "left": -25,
          "opacity": 1,
          "paddingVertical": 16,
          "position": "absolute",
          "top": 60,
          "transform": [
            {
              "translateY": 0,
            },
          ],
          "width": 750,
        }
      }
      testID="Hint"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          centerV={true}
          onLayout={[Function]}
          row={true}
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {},
              {},
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              [
                {
                  "backgroundColor": "#5A48F5",
                  "borderRadius": 20,
                  "maxWidth": 400,
                },
                {
                  "paddingBottom": 16,
                  "paddingHorizontal": 20,
                  "paddingTop": 12,
                },
                undefined,
                {
                  "backgroundColor": "#5A48F5",
                },
                false,
                {
                  "left": 0,
                },
              ],
            ]
          }
          testID="Hint.message"
        >
          <Text
            fsTagName="unmask"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "writingDirection": "ltr",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                [
                  {
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
            testID="Hint.message.text"
          >
            Hint message to hint things
          </Text>
        </View>
      </View>
      <Image
        accessibilityRole="image"
        accessible={false}
        assetGroup="icons"
        onError={[Function]}
        source={
          {
            "testUri": "../../../src/assets/internal/images/hintTipSide.png",
          }
        }
        style={
          [
            {
              "tintColor": "#5A48F5",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            false,
            [
              {
                "position": "absolute",
              },
              {
                "top": 9,
              },
              {
                "transform": [
                  {
                    "scaleY": 1,
                  },
                  {
                    "scaleX": 1,
                  },
                ],
              },
            ],
            false,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Hint Screen component test Different positions and scenarios left position should position correctly for a target positioned on left 1`] = `
<View
  centerV={true}
  padding-s5={true}
  style={
    [
      undefined,
      undefined,
      undefined,
      {
        "flex": 1,
      },
      undefined,
      {
        "padding": 20,
      },
      {},
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "backgroundColor": undefined,
        "height": 1334,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
    testID="Hint.overlay"
  />
  <View
    collapsable={true}
    message="Hint message to hint things"
    modifiers={
      {
        "alignments": {},
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "color": undefined,
        "flexStyle": undefined,
        "margins": {},
        "paddings": {},
        "positionStyle": undefined,
        "typography": undefined,
      }
    }
    position="bottom"
    style={
      [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {},
        {},
        {},
        [
          {
            "position": "absolute",
          },
          undefined,
          {
            "left": 25,
            "top": 370,
          },
          {
            "elevation": 10,
            "zIndex": 10,
          },
        ],
      ]
    }
    targetFrame={
      {
        "height": 60,
        "width": 100,
        "x": 25,
        "y": 370,
      }
    }
    useModal={true}
    visible={true}
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        {
          "alignItems": "flex-start",
          "left": -25,
          "opacity": 1,
          "paddingVertical": 16,
          "position": "absolute",
          "top": 60,
          "transform": [
            {
              "translateY": 0,
            },
          ],
          "width": 750,
        }
      }
      testID="Hint"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          centerV={true}
          onLayout={[Function]}
          row={true}
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {},
              {},
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              [
                {
                  "backgroundColor": "#5A48F5",
                  "borderRadius": 20,
                  "maxWidth": 400,
                },
                {
                  "paddingBottom": 16,
                  "paddingHorizontal": 20,
                  "paddingTop": 12,
                },
                undefined,
                {
                  "backgroundColor": "#5A48F5",
                },
                false,
                {
                  "left": 0,
                },
              ],
            ]
          }
          testID="Hint.message"
        >
          <Text
            fsTagName="unmask"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "writingDirection": "ltr",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                [
                  {
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
            testID="Hint.message.text"
          >
            Hint message to hint things
          </Text>
        </View>
      </View>
      <Image
        accessibilityRole="image"
        accessible={false}
        assetGroup="icons"
        onError={[Function]}
        source={
          {
            "testUri": "../../../src/assets/internal/images/hintTipMiddle.png",
          }
        }
        style={
          [
            {
              "tintColor": "#5A48F5",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            false,
            [
              {
                "position": "absolute",
              },
              {
                "top": 9,
              },
              {
                "transform": [
                  {
                    "scaleY": 1,
                  },
                  {
                    "scaleX": 1,
                  },
                ],
              },
            ],
            false,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Hint Screen component test Different positions and scenarios right position should position correctly for a target positioned on right - with sideTip 1`] = `
<View
  centerV={true}
  padding-s5={true}
  style={
    [
      undefined,
      undefined,
      undefined,
      {
        "flex": 1,
      },
      undefined,
      {
        "padding": 20,
      },
      {},
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "backgroundColor": undefined,
        "height": 1334,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
    testID="Hint.overlay"
  />
  <View
    collapsable={true}
    message="Hint message to hint things"
    modifiers={
      {
        "alignments": {},
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "color": undefined,
        "flexStyle": undefined,
        "margins": {},
        "paddings": {},
        "positionStyle": undefined,
        "typography": undefined,
      }
    }
    position="bottom"
    style={
      [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {},
        {},
        {},
        [
          {
            "position": "absolute",
          },
          undefined,
          {
            "left": 240,
            "top": 150,
          },
          {
            "elevation": 10,
            "zIndex": 10,
          },
        ],
      ]
    }
    targetFrame={
      {
        "height": 60,
        "width": 100,
        "x": 240,
        "y": 150,
      }
    }
    useModal={true}
    useSideTip={true}
    visible={true}
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        {
          "alignItems": "flex-start",
          "left": -240,
          "opacity": 1,
          "paddingVertical": 16,
          "position": "absolute",
          "top": 60,
          "transform": [
            {
              "translateY": 0,
            },
          ],
          "width": 750,
        }
      }
      testID="Hint"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          centerV={true}
          onLayout={[Function]}
          row={true}
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {},
              {},
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              [
                {
                  "backgroundColor": "#5A48F5",
                  "borderRadius": 20,
                  "maxWidth": 400,
                },
                {
                  "paddingBottom": 16,
                  "paddingHorizontal": 20,
                  "paddingTop": 12,
                },
                undefined,
                {
                  "backgroundColor": "#5A48F5",
                },
                false,
                {
                  "left": 0,
                },
              ],
            ]
          }
          testID="Hint.message"
        >
          <Text
            fsTagName="unmask"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "writingDirection": "ltr",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                [
                  {
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
            testID="Hint.message.text"
          >
            Hint message to hint things
          </Text>
        </View>
      </View>
      <Image
        accessibilityRole="image"
        accessible={false}
        assetGroup="icons"
        onError={[Function]}
        source={
          {
            "testUri": "../../../src/assets/internal/images/hintTipSide.png",
          }
        }
        style={
          [
            {
              "tintColor": "#5A48F5",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            false,
            [
              {
                "position": "absolute",
              },
              {
                "top": 9,
              },
              {
                "transform": [
                  {
                    "scaleY": 1,
                  },
                  {
                    "scaleX": 1,
                  },
                ],
              },
            ],
            false,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Hint Screen component test Different positions and scenarios right position should position correctly for a target positioned on right 1`] = `
<View
  centerV={true}
  padding-s5={true}
  style={
    [
      undefined,
      undefined,
      undefined,
      {
        "flex": 1,
      },
      undefined,
      {
        "padding": 20,
      },
      {},
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "backgroundColor": undefined,
        "height": 1334,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
    testID="Hint.overlay"
  />
  <View
    collapsable={true}
    message="Hint message to hint things"
    modifiers={
      {
        "alignments": {},
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "color": undefined,
        "flexStyle": undefined,
        "margins": {},
        "paddings": {},
        "positionStyle": undefined,
        "typography": undefined,
      }
    }
    position="bottom"
    style={
      [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {},
        {},
        {},
        [
          {
            "position": "absolute",
          },
          undefined,
          {
            "left": 240,
            "top": 150,
          },
          {
            "elevation": 10,
            "zIndex": 10,
          },
        ],
      ]
    }
    targetFrame={
      {
        "height": 60,
        "width": 100,
        "x": 240,
        "y": 150,
      }
    }
    useModal={true}
    visible={true}
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        {
          "alignItems": "flex-start",
          "left": -240,
          "opacity": 1,
          "paddingVertical": 16,
          "position": "absolute",
          "top": 60,
          "transform": [
            {
              "translateY": 0,
            },
          ],
          "width": 750,
        }
      }
      testID="Hint"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          centerV={true}
          onLayout={[Function]}
          row={true}
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {},
              {},
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              [
                {
                  "backgroundColor": "#5A48F5",
                  "borderRadius": 20,
                  "maxWidth": 400,
                },
                {
                  "paddingBottom": 16,
                  "paddingHorizontal": 20,
                  "paddingTop": 12,
                },
                undefined,
                {
                  "backgroundColor": "#5A48F5",
                },
                false,
                {
                  "left": 0,
                },
              ],
            ]
          }
          testID="Hint.message"
        >
          <Text
            fsTagName="unmask"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "writingDirection": "ltr",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                [
                  {
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
            testID="Hint.message.text"
          >
            Hint message to hint things
          </Text>
        </View>
      </View>
      <Image
        accessibilityRole="image"
        accessible={false}
        assetGroup="icons"
        onError={[Function]}
        source={
          {
            "testUri": "../../../src/assets/internal/images/hintTipMiddle.png",
          }
        }
        style={
          [
            {
              "tintColor": "#5A48F5",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            false,
            [
              {
                "position": "absolute",
              },
              {
                "top": 9,
              },
              {
                "transform": [
                  {
                    "scaleY": 1,
                  },
                  {
                    "scaleX": 1,
                  },
                ],
              },
            ],
            false,
          ]
        }
      />
    </View>
  </View>
</View>
`;
